apps/dashboard/app/models/batch_connect/session.rb

```diff
@@ -556,11 +556,26 @@ def connection_in_info?
     # Whether to allow SSH to node running the job
     # @return [Boolean] whether to allow SSH to node running the job
     def ssh_to_compute_node?
-      if !cluster_ssh_to_compute_node?.nil?
-        return cluster_ssh_to_compute_node?
-      else
-        return global_ssh_to_compute_node?
-      end
+      return cluster_and_app_ssh_to_compute_node? if override_global_ssh_to_compute_node?
+      global_ssh_to_compute_node?
+    end
+
+    def cluster_and_app_ssh_to_compute_node?
+      cluster_ssh_to_compute_node? && app_ssh_to_compute_node?
+    end
+
+    def override_global_ssh_to_compute_node?
+      cluster_override_ssh_to_compute_node? || app_override_ssh_to_compute_node?
+    end
+
+    def cluster_override_ssh_to_compute_node?
+      !cluster_ssh_to_compute_node?.nil? &&
```

Comment:
I'm not in love with the `?.` here. I'm guessing the actual convention for `?` is that it guarantees a boolean, with these now able to return nil, I'm thinking we should treat it like any other API requesting a value instead of asking a question.

That doesn't apply to all of these, like `override_global_ssh_to_compute_node?` above, but that does seem to guarantee a boolean return value.